gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -807,10 +845,14 @@ private void beforeName() throws IOException {
   private void beforeValue() throws IOException {
     switch (peek()) {
       case NONEMPTY_DOCUMENT:
-        if (strictness != Strictness.LENIENT) {
-          throw new IllegalStateException("JSON must have only one top-level value.");
+        if (topLevelSeparator == null) {
+          throw new IllegalStateException(
+              "Multiple top-level values support has not been enabled, use"
+                  + " `JsonWriter.setTopLevelSeparator(String)`");
         }
-        // fall-through
```

Comment:
Looks like this "fall-through" was previously redundant: The next case is `EMPTY_DOCUMENT` which replaces the top of the stack with `NONEMPTY_DOCUMENT`, but here the top is already `NONEMPTY_DOCUMENT`.